src/Language/Haskell/Liquid/Parse.hs

```diff
@@ -750,13 +749,18 @@ tyBodyP ty
           outTy (RFun _ _ t _) = Just t
           outTy _              = Nothing
 
+upperIdP' :: Parser Symbol
+upperIdP' = try $ symbol <$> condIdP alphanums (isUpper . head)
+  where
+    alphanums = ['A' .. 'Z'] ++ ['a'..'z'] ++ ['0'..'9'] ++ "'"
```

Comment:
`isUpper` would prevent that, but the point about `'` in identifiers is still true.